src/main/java/de/tum/p2p/api/netty/handle/OnionTunnelIncomingMessageHandler.java

```diff
@@ -0,0 +1,20 @@
+package de.tum.p2p.api.netty.handle;
+
+import de.tum.p2p.api.netty.forwarding.OnionTunnelIncomingMessage;
+import de.tum.p2p.onion.forwarding.OnionForwarder;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+
+public class OnionTunnelIncomingMessageHandler extends SimpleChannelInboundHandler<OnionTunnelIncomingMessage> {
+
+    private OnionForwarder onionForwarder;
+
+    public OnionTunnelIncomingMessageHandler(OnionForwarder onionForwarder) {
+        this.onionForwarder = onionForwarder;
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, OnionTunnelIncomingMessage msg) throws Exception {
+//        onionForwarder..addIncomingTunnelObserver();
```

Comment:
OnionTunnelIncomingMessage is outbound not inbound message. You should do this way (in OnionForwarderAPI):
```
onionForwarder.addIncomingTunnelObserver((tunnelId) -> {
 channel.writeAndFlush(new OnionTunnelIncoming(...));
});
```
 where the `channel` is a channel you'll get as a result of this line `b.bind(port).sync().channel().closeFuture().sync();`